libraries/botbuilder-core/tests/cloudAdapterBase.test.js

```diff
@@ -0,0 +1,435 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const assert = require('assert');
+const sinon = require('sinon');
+const { CloudAdapterBase, TurnContext, ActivityTypes, INVOKE_RESPONSE_KEY } = require('..');
+
+const {
+    BotFrameworkAuthenticationFactory,
+    ClaimsIdentity,
+    AuthenticationConstants,
+} = require('botframework-connector');
+
+class TestAdapter extends CloudAdapterBase {}
+
+const noop = () => null;
+
+describe('CloudAdapterBase', function () {
+    let sandbox;
+    beforeEach(function () {
+        sandbox = sinon.createSandbox({ useFakeTimers: true });
+    });
+
+    afterEach(function () {
+        if (sandbox) {
+            sandbox.restore();
+        }
+    });
+
+    const authentication = BotFrameworkAuthenticationFactory.create();
+    const adapter = new TestAdapter(authentication);
+
+    const mockConnectorClient = (context) => {
+        const connectorClient = {
+            conversations: {
+                createConversation: noop,
+                deleteActivity: noop,
+                replyToActivity: noop,
+                sendToConversation: noop,
+                updateActivity: noop,
+            },
+        };
+
+        if (context) {
+            context.turnState.set(context.adapter.ConnectorClientKey, connectorClient);
+        }
+
+        const mock = sandbox.mock(connectorClient.conversations);
+
+        return { connectorClient, mock };
+    };
+
+    describe('constructor', function () {
+        it('throws for bad args', function () {
+            assert.throws(() => new TestAdapter(), { name: 'TypeError' });
+        });
+
+        it('succeeds', function () {
+            new TestAdapter(BotFrameworkAuthenticationFactory.create());
+        });
+    });
+
+    describe('sendActivities', function () {
+        it('throws for bad args', function () {
+            assert.throws(() => adapter.sendActivities(undefined, []), { name: 'TypeError' });
+            assert.throws(() => adapter.sendActivities(new TurnContext(adapter), undefined), { name: 'TypeError' });
+            assert.throws(() => adapter.sendActivities(new TurnContext(adapter), []), { name: 'Error' });
```

Comment:
nit: Should these `assert.throws()` also assert the correct error message? An edge case scenario where this might be valuable to check for would be a dev removing the argument check and the test throwing downstream for a separate reason, and then inappropriately passing.

Reply:
Good call!